app/presenters/hyrax/displays_image.rb

```diff
@@ -13,17 +13,18 @@ def display_image
       return nil unless ::FileSet.exists?(id) && solr_document.image? && current_ability.can?(:read, id)
       # @todo this is slow, find a better way (perhaps index iiif url):
       original_file = ::FileSet.find(id).original_file
+      latest_file_id = original_file.has_versions? ? ActiveFedora::File.uri_to_id(original_file.versions.last.uri) : original_file.id
```

Comment:
Also, `original_file.has_versions?` always seems to return true even when a file_set has only one version, so our `iiif_endpoint`s changed for all our manifests, invalidating Rails.cache entries. Perhaps something like this is needed?:
```
latest_file_id = original_file.versions.count > 1 ? ActiveFedora::File.uri_to_id(original_file.versions.last.uri) : original_file.id
```

Reply:
The use of has_versions? was meant as a safeguard in case a file did not have support for versioning, although I do not know if that is really an issue in any application.

For now I'm working on the potential timestamp based solution.

Reply:
Thanks for the knowledge. And thanks for getting this solution going so we could begin applying it to Sufia. 

Not sure if it helps to share, but we are experiencing some older version Riiif tiles still when a new file_set version derivative is not completed yet. `FileSetPresenter#display_image` returns the new `iiif_endpoint` using the new Fedora file version id, but when the images requests are sent to Riiif, the new file id invalidates the Rails.cache keys, sparking the file resolver to check it's cache for a matching hashed file name, but it's checking the derivative path and the timestamp of the *old* version. That new file can sometimes not be ready yet, since it is created in a background job.

What i plan to do is control the `iiif_endpoint` change from some sort of `after_perform` on the `CreateDerivativesJob` - that way the derivative file will be ready when the `HTTPFileResolver` fetches.

What I'm not 100% sure of yet is how I will control the versioning. I may just increment some solr field on the file_set that acts as the version.

Any thoughts from the community are welcome, even those that correct my thinking. All of this is new to me and I may be a smidgeon off.

Best,
Kevin